api/controllers/game/counter.js

```diff
@@ -15,14 +15,15 @@ module.exports = function (req, res) {
           if (card.rank === 2) {
             if (!opHasQueen) {
               if (game.twos.length > 0) {
-                logEntry = `${userService.truncateEmail(player.email)} played the ${card.name} to counter ${userService.truncateEmail(opponent.email)}'s ${game.twos[game.twos.length - 1].name}.`;
+                logEntry = `${player.username} played the ${card.name} to counter ${opponent.username}'s ${game.twos[game.twos.length - 1].name}.`;
               } else {
-                logEntry = `${userService.truncateEmail(player.email)} played the ${card.name} to counter ${userService.truncateEmail(opponent.email)}'s ${game.oneOff.name}.`;
+                logEntry = `${player.username} played the ${card.name} to counter ${opponent.username}'s ${game.oneOff.name}.`;
               }
               const gameUpdates = {
                 lastEvent: {
                   change: 'counter',
                   pNum: req.session.pNum,
+                  log: [game.log, ...logEntry]
```

Comment:
Think this was a bug

Reply:
Good catch!

Reply:
But the spread operator is backwards. It should be:
```
log: [...game.log, logEntry]
```

`...game.log` unpacks the array into its constituent elements so generally `[...someArray]`, makes a copy of `someArray` with the exact same elements, and `[...someArray, newElement]` creates a copy of `someArray` that also has `newElement` at the end of it, which is what we want for the game log

Reply:
dope!  yep, fixed